### PR TITLE
Added bot setup guide to README, improved texts on commands, random code fixes and improvements

### PR DESCRIPTION
**Pull request by m1ksu**
Added bot setup guide to README, improved texts on commands, random code fixes and improvements

### IMPORTANT SIDENOTE: 
CHANGED BOT'S CODE TO TAKE A KEY NAMED "token" FROM THE CONFIG FILE INSTEAD OF CALLING "getToken()"**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,9 @@
-src/config.js
+data/config.json
+data/config.js
 node_modules/
+run.bat
```
Comment (Jackojc) [inline .gitignore:4]: Why are you ignoring the file that runs the bot?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,58 @@
 # HopsonBot
-Discord bot for the Hopson Community Server, written using Discord.js
+HopsonBot is a Discord bot for the [Hopson Community Server](https://discordapp.com/invite/xvQaysR), written using Discord.js.
 
-> Put your config in the data folder.
```
Comment (Jackojc) [inline README.md:4]: I don't think the bot is being used by any other server for now or in the foreseeable future.

File: src/command_handler.js [hunk 3]
```diff
@@ -86,27 +86,26 @@ function addFunctionCommand(name, func, description, acceptsArgs)
...
+    "Displays this list of commands.",
     false
 );
 
-
 addFunctionCommand(
     "role",
     RoleMod.tryModifyRole,
-    "Allows the user to add or remove role(s) from '>rolelist'\nUseage: '>role add C++ Java'",
+    "Allows the user to give or remove \"skill\" roles from themselves. Available roles are displayed in **`>rolelist`**.\nUsage example: **`>role add C++ Java`** & **`>role remove Rust`**",
```
Comment (Jackojc) [inline src/command_handler.js:109]: This is phrased weirdly, the original was fine.

File: .gitignore (hunk 1 above)
Comment (Hopson97) [reply .gitignore:4]: Agreed, it shoudn't be ignored

File: data/config.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "token": "Mzk0ODIyMjE0MTYxMDA2NTky.DSUT7A.jIaQ4Cx-Oysl46VS2EzvegnYQ14"
+}
```
Comment (Jackojc) [inline data/config.json:3]: Um...

**Comment by Jackojc** [comment]
Please resubmit some of your changes as individual PRs and fix any comments that may have been made.

**Pull request closed without merging**